### DISCUSSION TITLE
How to access fields of a component from a BlockPostObject component?

### DISCUSSION THREAD

**Discussion by marcotiberio** [flyntwp/flynt#421] [Q&A] [answered]
Hello,
I'm trying to display a few fields from a component that is displaying info on a post (i.e. artist bio, artist social media, etc) in a BlockPostObject component. If I'm not mistaking I should be able to access that specific component structure via:
`{{ renderComponent('BlockArtistInfo', { bio: bioHtml, image: post.thumbnail, options: { size: 'sizeLarge' } }) }}` but I can't rightfully get the "bioHtml" field to show for example.

And if I try to display any content via `{{ item.post_content|e('wp_kses_post') }}` of course get the `[flyntTheContent id="37"]`.

I definitely miss something and I can't see. So basically my question is, how can I access specific components and their fields from another `relationship` or `post_object` component without these fields being stored in the `meta` of the post?

I hope this is clear enough :)

Thank you for your help!

Marco

**Comment by domtra** [comment, marked as answer]
Hi @marcotiberio , 
I do not get a full picture of your setup or what you are trying to do exactly. So I will make a couple of assumptions and try to provide a solution. Please correct if I got something wrong.

Let's say you have a page. On that page is a flexible content field. You have a component BlockArtistInfo in which you select a different post, maybe of post type artist through a ACF relationship or post_object field. And for this connected post, you want to access the meta fields.

The challenge is that when you use `'return_format' => 'object'` in ACF's relation fields, it returns a regular WordPress post object. Not a TimberPost. So you cannot use the methods you are used to with Timber/Twig.

What we usually do in the `addComponentData` filter of the component is to convert the regular post object to a timber post. Something like 

```php
add_filter('Flynt/addComponentData?name= BlockArtistInfo', function ($data) {
    $data['item'] = new \Timber\Post($data['item']);
    return $data;
});
```

if item is the relation field name.

Then you can do the following it you twig file 
```twig
{{ item.meta('bioHtml') }}
```
or something similar.

Let me know if that helps you.
